server/service/apple_mdm.go

```diff
@@ -2946,6 +2946,13 @@ func ReconcileAppleProfiles(
 			continue
 		}
 
+		if p.Status != nil && *p.Status == fleet.MDMDeliveryFailed && p.OperationType == fleet.MDMOperationTypeInstall {
```

Comment:
seems like this is the same condition as the one in `server/datastore/mysql/apple_mdm.go`? since it's fairly complex, should we abstract it? maybe to whatever is the type of `p`?